tests/components/mqtt/test_config_flow.py

```diff
@@ -183,15 +183,14 @@ async def test_user_connection_works(
     assert result["type"] == "form"
 
     result = await hass.config_entries.flow.async_configure(
-        result["flow_id"], {"broker": "127.0.0.1", "advanced_options": False}
+        result["flow_id"], {"broker": "127.0.0.1"}
     )
 
     assert result["type"] == "create_entry"
     assert result["result"].data == {
         "broker": "127.0.0.1",
         "port": 1883,
         "discovery": True,
-        "discovery_prefix": "homeassistant",
```

Comment:
Default settings are not added to the entry by default, they are added during entry load by applying the schema.